### PR TITLE
Task 5.1 Sparse matrix

### PR DESCRIPTION
**Pull request by stoyankisyov**

### CODE REVIEW

File: SparseMatrix/SparseMatrix.cs [hunk 1]
```diff
@@ -0,0 +1,114 @@
+﻿using System.Collections;
+
+namespace SparseMatrixApplication
+{
+    public class SparseMatrix : IEnumerable<long>
+    {
```
Comment (IgorKheidorov) [inline SparseMatrix/SparseMatrix.cs:7]: Has to be properties with public get

File: SparseMatrix/SparseMatrix.cs [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+                }
+                else
+                {
+                    _elements.Remove((i, j));
+                }
+            }
+        }
+
+        public override string ToString()
+        {
```
Comment (IgorKheidorov) [inline SparseMatrix/SparseMatrix.cs:56]: Use StringBuilder here for effectiveness

File: SparseMatrix/SparseMatrix.cs [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public IEnumerable<(int, int, long)> GetNonzeroElements()
+        {
```
Comment (IgorKheidorov) [inline SparseMatrix/SparseMatrix.cs]: Use linq for the whole result without foreach   (tip-> use Select to form tuple from dictionary)

File: SparseMatrix/SparseMatrix.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+                .ToDictionary();
+
+            foreach (var keyValuePair in sortedElementsByColumn)
+            {
+                yield return (keyValuePair.Key.Item1, keyValuePair.Key.Item2, keyValuePair.Value);
+            }
+        }
+
+        public int GetCount(long element)
+        {
```
Comment (IgorKheidorov) [inline SparseMatrix/SparseMatrix.cs]: use ternary operator for return

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Totally good

File: SparseMatrix/SparseMatrix.cs (hunk 1 above)
Comment (stoyankisyov) [reply SparseMatrix/SparseMatrix.cs:7]: Fixed

File: SparseMatrix/SparseMatrix.cs (hunk 2 above)
Comment (stoyankisyov) [reply SparseMatrix/SparseMatrix.cs:56]: Fixed

File: SparseMatrix/SparseMatrix.cs (hunk 3 above)
Comment (stoyankisyov) [reply SparseMatrix/SparseMatrix.cs]: Fixed

File: SparseMatrix/SparseMatrix.cs (hunk 4 above)
Comment (stoyankisyov) [reply SparseMatrix/SparseMatrix.cs]: Fixed

**Review by IgorKheidorov — APPROVED** [review]

**Pull request merged**